src/main/webapp/app/tasks/task-list/task-list.component.html

```diff
@@ -1,9 +1,24 @@
-<mat-list data-cy="task-list">
-  <mat-list-item *ngFor="let task of tasks" class="mat-elevation-z1">
-    <mat-icon mat-list-icon>assignment</mat-icon>
+<div class="task-list-menu mat-elevation-z1">
+
+  <button mat-icon-button aria-label="Delete all done tasks" color="primary" matTooltip="Delete all done tasks">
+    <mat-icon color="warn" aria-label="Delete all done tasks" (click)="deleteAllDone(task)">remove_done</mat-icon>
+  </button>
+</div>
+
+
+<mat-list cdkDropList (cdkDropListDropped)="drop($event)" data-cy="task-list">
+  <mat-list-item *ngFor="let task of tasks" class="mat-elevation-z1 task-list-item" [ngClass]="{'task-done': task.done}"
+    cdkDrag>
+    <mat-icon mat-list-icon *ngIf="task.done">assignment_turned_in</mat-icon>
+    <mat-icon mat-list-icon *ngIf="!task.done">assignment</mat-icon>
     <h4 mat-line>{{task.name}}</h4>
+    <button mat-icon-button aria-label="Toggle status" color="primary">
+      <mat-icon aria-label="Toggle task status" (click)="toggleStatus(task)" *ngIf="!task.done">check_box_outline_blank
```

Comment:
as you only show the icon on !task.done, there will be still be the button present and a hover effect when you mouse over it, would rather do this on the button

Reply:
You mean "delete all done" button? Ah yes, this button may be removed once clicked until there is a new "done" task. But no big harm and there is no extra request sent to the backend.

Reply:
There is room to improve the behavior for sure. Here there is no harm to have the "delete all done" button but for sure, this can be improved as you said.

Reply:
I was just reviewing the code again. I just noticed that *ngIf is on the icon. It should be on the button of course.

Reply:
Btw, I am not used to GitHub but rather Gitlab. That's why I did not understand where your comments point to, at the first sight.